### PR TITLE
executor/linux: make chroot binary paths absolute

### PR DESCRIPTION
**Pull request by schmichael**
Avoid libcontainer.Process trying to lookup the binary via $PATH as the
executor has already found where the binary is located.

### CODE REVIEW

File: drivers/shared/executor/executor_linux.go [hunk 1]
```diff
@@ -161,7 +161,10 @@ func (l *LibcontainerExecutor) Launch(command *ExecCommand) (*ProcessState, erro
 	if err != nil {
 		return nil, fmt.Errorf("failed to determine relative path base=%q target=%q: %v", command.TaskDir, path, err)
 	}
-	path = rel
+
+	// Turn relative-to-chroot path into absolute path to avoid
+	// libcontainer trying to resolve the binary using $PATH
+	path = "/" + rel
```
Comment (notnoop) [inline drivers/shared/executor/executor_linux.go]: We probably should check here if `rel` starts with `../` and report an error with binary being outside libcontainer root.

File: client/allocrunner/taskrunner/task_runner_test.go [hunk 2]
```diff
@@ -1054,6 +1054,85 @@ func TestTaskRunner_DeriveToken_Unrecoverable(t *testing.T) {
...
+	alloc := mock.BatchAlloc()
+	alloc.Job.TaskGroups[0].RestartPolicy = &structs.RestartPolicy{}
+	task := alloc.Job.TaskGroups[0].Tasks[0]
+	task.Driver = "exec"
+	task.Config = map[string]interface{}{
+		"command": "noop.sh",
+	}
+	task.Artifacts = []*structs.TaskArtifact{
+		{
+			GetterSource: fmt.Sprintf("%s/testdata/noop.sh", ts.URL),
```
Comment (preetapan) [inline client/allocrunner/taskrunner/task_runner_test.go]: cool way of setting up an artifact source for a test!

File: drivers/shared/executor/executor_linux.go (hunk 1 above)
Comment (preetapan) [inline drivers/shared/executor/executor_linux.go]: Do we need to handle the case when rel already starts with "/" ?

Comment (nickethier) [inline drivers/shared/executor/executor_linux.go]: Should we use `filepath.Join` here?

Comment (schmichael) [reply drivers/shared/executor/executor_linux.go]: Going to handle it by using filepath.Join at @nickethier's suggestion below.

**Pull request merged**